### PR TITLE
LPS-151836 Test Automation Creation | master

### PR DESCRIPTION
**Pull request by ozysouza**
https://issues.liferay.com/browse/LPS-151836

cc @david-gutierrez-mesa 
cc @Tim-Cao

### CODE REVIEW

**Comment by Tim-Cao** [comment]
Hi @ozysouza, would you mind updating the cases according to comments from https://github.com/liferay-echo/liferay-portal/pull/7917/commits/dd6fecfcf0f0fb8617ade48899fb426c697d7f65
Thanks

**Comment by ozysouza** [comment]
> Hi @ozysouza, would you mind updating the cases according to comments from [dd6fecf](https://github.com/liferay-echo/liferay-portal/commit/dd6fecfcf0f0fb8617ade48899fb426c697d7f65) Thanks

Hi @Tim-Cao, there is no comment there. What do you mean?

**Comment by Tim-Cao** [comment]
Sorry I forgot to click review button

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/stylebooks/StyleBooks.testcase [hunk 1]
```diff
@@ -622,4 +622,34 @@ definition {
...
+		task ("Then the Style books designer can see the new icons in the dropdown menu") {
+			MenuItem.viewIcon(iconList = "pencil,change,upload,copy,trash");
+		}
+
+		task ("When the Style books designer check the style book checkbox") {
+			LexiconCard.checkCardCheckbox(card = "Test Style Book Name");
+		}
+
+		task ("Then the Style books designer can see the new icons in the management toolbar") {
+			ManagementBar.viewIcon(iconList = "upload,copy,trash");
```
[the remark this answers is not in the document]
Comment (ozysouza) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/stylebooks/StyleBooks.testcase:651]: I going to move the macro that I created on MenuItem.viewIcon to DropdownMenuItem.viewIcon, I also noticed I was verifying the icon on the wrong place, it should be in the dropdown menu of managment bar, but I'll keep the macro that I created on ManagementBar.viewIcon and add a step to also verify the icons on the dropdown. What do you think?

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/stylebooks/StyleBooks.testcase [hunk 2]
```diff
@@ -622,4 +622,34 @@ definition {
...
+
+		task ("Then the Style books designer can see the new icons in the dropdown menu") {
+			MenuItem.viewIcon(iconList = "pencil,change,upload,copy,trash");
+		}
+
+		task ("When the Style books designer check the style book checkbox") {
+			LexiconCard.checkCardCheckbox(card = "Test Style Book Name");
+		}
+
+		task ("Then the Style books designer can see the new icons in the management toolbar") {
```
[the remark this answers is not in the document]
Comment (ozysouza) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/stylebooks/StyleBooks.testcase:650]: Do you mind if I let the message as "Then the Style books designer can see the new icons in the management toolbar and dropdown menu". I would like to verify the icon on both, managment toolbar and dropdown menu.

**Comment by ozysouza** [comment]
Hi @Tim-Cao, I've updated the PR. Would you have a look at? Thanks!

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/FragmentsAdmin.testcase [hunk 3]
```diff
@@ -661,4 +661,61 @@ definition {
...
+			task ("Add a Fragment Set") {
+				JSONFragment.addFragmentCollection(
+					groupName = "Test Site Name",
+					name = "New Fragment Set Name");
+			}
+
+			task ("Add a fragment") {
+				JSONFragment.addSection(
+					collectionName = "New Fragment Set Name",
+					groupName = "Test Site Name",
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/FragmentsAdmin.testcase:677]: Add	var fragmentStatus = "Approved";

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/FragmentsAdmin.testcase [hunk 4]
```diff
@@ -661,4 +661,61 @@ definition {
...
+					name = "Fragment Name");
+			}
+		}
+
+		task ("When the Fragments creator open the ellipsis dropdown menu of custom fragment") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/FragmentsAdmin.testcase:682]: When the Fragments creator open the dropdown menu of custom fragment

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/FragmentsAdmin.testcase [hunk 5]
```diff
@@ -661,4 +661,61 @@ definition {
...
+			Click.waitForMenuToggleJSClick(
+				key_card = "Fragment Name",
+				locator1 = "Card#CARD_ELLIPSIS");
+		}
+
+		task ("Then the Fragments creator can see the new icons on the dropdown menu of custom fragment") {
+			DropdownMenuItem.viewIcon(iconList = "pencil,change,upload,copy,move-folder,trash");
+		}
+
+		task ("When the Fragments creator open the ellipsis dropdown menu of custom fragment set") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/FragmentsAdmin.testcase:694]: When the Fragments creator open the dropdown menu of custom fragment set

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/FragmentsAdmin.testcase [hunk 6]
```diff
@@ -661,4 +661,61 @@ definition {
...
+			Click.waitForMenuToggleJSClick(locator1 = "Icon#BODY_VERTICAL_ELLIPSIS");
+		}
+
+		task ("Then the Fragments creator can see the new icons on the dropdown menu of custom fragment set") {
+			DropdownMenuItem.viewIcon(iconList = "pencil,upload,import,trash");
+		}
+
+		task ("When the Fragments creator open the ellipsis dropdown menu on the Fragment Sets") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/fragment/FragmentsAdmin.testcase:702]: When the Fragments creator open the dropdown menu on the Fragment Sets

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/contentpage/ContentPages.testcase [hunk 7]
```diff
@@ -1084,6 +1084,57 @@ definition {
...
+					pageName = "Test Content Page Name",
+					siteName = "Test Site Name");
+
+				PageEditor.addFragment(
+					collectionName = "Content Display",
+					fragmentName = "Collection Display");
+			}
+		}
+
+		task ("When the Page designer select Collection Display and open its ellipsis dropdown menu") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/contentpage/ContentPages.testcase:1110]: When the Page designer open the dropdown menu of Collection Display topper

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/stylebooks/StyleBooks.testcase [hunk 8]
```diff
@@ -622,4 +622,36 @@ definition {
...
+	@description = "This is a test for LPS-145292. The user could view new icons on dropdown menu of style book."
+	@priority = "4"
+	test ViewDropdownMenuIconsInStyleBooksAdmin {
+		task ("Given a Style books designer has a new style book") {
+			JSONStylebooks.addStyleBookEntry(
+				groupName = "Test Site Name",
+				styleBookEntryName = "Test Style Book Name");
+		}
+
+		task ("When the Style books designer opens the ellipsis dropdown menu of style book") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/stylebooks/StyleBooks.testcase:634]: When the Style books designer opens the dropdown menu of style book

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/sitenavigation/navigationmenu/NavigationMenus.testcase [hunk 9]
```diff
@@ -1109,6 +1109,36 @@ definition {
...
+	@description = "This is a test for LPS-145292. The user could view new icons on dropdown menu of navigation menu."
+	@priority = "4"
+	test ViewDropdownMenuIconsInNavigationAdmin {
+		task ("Given a Navigation menu creator has a new navigation menu") {
+			JSONSitenavigation.addSiteNavigationMenu(
+				groupName = "Test Site Name",
+				siteNavigationMenuName = "Navigation Menu Name");
+		}
+
+		task ("When the Navigation menu creator open the ellipsis dropdown menu of navigation menu") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/sitenavigation/navigationmenu/NavigationMenus.testcase:1121]: When the Navigation menu creator open the dropdown menu of navigation menu

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/TemplatesAdmin.testcase [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+			task ("Add a widget template for Breadcrumb widget") {
+				ApplicationDisplayTemplates.openWidgetTemplatesAdmin(siteURLKey = "test-site-name");
+
+				ApplicationDisplayTemplates.addCP(
+					adtType = "Breadcrumb",
+					fieldNameList = "Breadcrumb Entry");
+			}
+		}
+
+		task ("When the Template creator opens the ellipsis dropdown menu of widget template") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/TemplatesAdmin.testcase:76]: When the Template creator opens the dropdown menu of widget template

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/TemplatesAdmin.testcase [hunk 11]
```diff
@@ -0,0 +1,97 @@
...
+
+			Click(locator1 = "ManagementBar#MORE_ACTIONS_ELLIPSIS");
+		}
+
+		task ("Then the Template creator can see the new icon on the dropdown menu") {
+			DropdownMenuItem.viewIcon(iconList = "trash");
+		}
+
+		task ("Given a Template creator has a Widget Template") {
+			task ("Add a widget template for Breadcrumb widget") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/TemplatesAdmin.testcase:67]: No need task block

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/TemplatesAdmin.testcase [hunk 12]
```diff
@@ -0,0 +1,97 @@
...
+		else {
+			JSONGroup.deleteGroupByName(groupName = "Test Site Name");
+		}
+	}
+
+	@description = "This is a test for LPS-145292. The user could view new icons on dropdown menu of template."
+	@priority = "4"
+	test ViewDropdownMenuIconsInTemplatesAdmin {
+		task ("Given a Template creator has an Information Template") {
+			task ("Add an information template for Blogs Entry") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/TemplatesAdmin.testcase:35]: No need task block

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/TemplatesAdmin.testcase [hunk 13]
```diff
@@ -0,0 +1,97 @@
...
+				InformationTemplates.openInformationTemplatesAdmin(siteURLKey = "test-site-name");
+
+				InformationTemplates.addCP(
+					informationTemplateName = "Test Information Template",
+					itemType = "Blogs Entry");
+			}
+		}
+
+		task ("When the Template creator open the ellipsis dropdown menu of information template") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/TemplatesAdmin.testcase:44]: When the Template creator open the dropdown menu of information template

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/PageTemplatesAdmin.testcase [hunk 14]
```diff
@@ -48,4 +48,147 @@ definition {
...
+
+			Click(locator1 = "ManagementBar#MORE_ACTIONS_ELLIPSIS");
+		}
+
+		task ("Then the Page Template creator can see the new icons on the dropdown menu") {
+			DropdownMenuItem.viewIcon(iconList = "upload,trash");
+		}
+
+		task ("Given a Page Template creator has a new Masters Template") {
+			task ("Add a custom master") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/PageTemplatesAdmin.testcase:158]: No need task block

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/PageTemplatesAdmin.testcase [hunk 15]
```diff
@@ -48,4 +48,147 @@ definition {
...
+
+			Click(locator1 = "ManagementBar#MORE_ACTIONS_ELLIPSIS");
+		}
+
+		task ("Then the Page Template creator can see the new icon on the dropdown menu") {
+			DropdownMenuItem.viewIcon(iconList = "upload,trash");
+		}
+
+		task ("Given a Page Template creator has a new Page Template Set with a Content and Widget Page Template") {
+			task ("Add a Page Template Sets") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/PageTemplatesAdmin.testcase:91]: No need task block

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/PageTemplatesAdmin.testcase [hunk 16]
```diff
@@ -48,4 +48,147 @@ definition {
...
+				JSONLayoutpagetemplate.addLayoutPageTemplateCollection(
+					groupName = "Test Site Name",
+					layoutPageTemplateCollectionName = "Test Page Template Set");
+			}
+
+			task ("Add a content and widget page template") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/PageTemplatesAdmin.testcase:97]: No need task block

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/PageTemplatesAdmin.testcase [hunk 17]
```diff
@@ -48,4 +48,147 @@ definition {
...
+						groupName = "Test Site Name",
+						layoutPageTemplateCollectionName = "Test Page Template Set",
+						layoutPageTemplateEntryName = "${type} Template Name",
+						layoutPageTemplateStatus = "Approved",
+						layoutPageTemplateType = "${type} Page Template");
+				}
+			}
+		}
+
+		task ("When the Page Template creator open the ellipsis dropdown menu of Page Template Sets") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/PageTemplatesAdmin.testcase:109]: When the Page Template creator open the dropdown menu of Page Template Sets

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/PageTemplatesAdmin.testcase [hunk 18]
```diff
@@ -48,4 +48,147 @@ definition {
...
+
+		task ("When the Page Template creator open the ellipsis dropdown menu of custom page template set") {
+			Click.waitForMenuToggleJSClick(locator1 = "Icon#BODY_VERTICAL_ELLIPSIS");
+		}
+
+		task ("Then the Page Template creator can see the new icons on the dropdown menu") {
+			DropdownMenuItem.viewIcon(iconList = "pencil,password-policies,trash");
+		}
+
+		task ("When the Page Template creator open the ellipsis dropdown menu of content template") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/PageTemplatesAdmin.testcase:127]: When the Page Template creator open the dropdown menu of content page template

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/PageTemplatesAdmin.testcase [hunk 19]
```diff
@@ -48,4 +48,147 @@ definition {
...
+			Click.waitForMenuToggleJSClick(
+				key_card = "Content Template Name",
+				locator1 = "Card#CARD_ELLIPSIS");
+		}
+
+		task ("Then the Page Template creator can see the new icons on the dropdown menu") {
+			DropdownMenuItem.viewIcon(iconList = "pencil,move-folder,change,upload,cog,password-policies,trash");
+		}
+
+		task ("When the Page Template creator open the ellipsis dropdown menu of widget template") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/PageTemplatesAdmin.testcase:137]: When the Page Template creator open the dropdown menu of widget page template

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/PageTemplatesAdmin.testcase [hunk 20]
```diff
@@ -48,4 +48,147 @@ definition {
 		}
 	}
 
+	@description = "This is a test for LPS-145292. The user could view new icons on dropdown menu of page template."
+	@priority = "4"
+	test ViewDropdownMenuIconsInPageTemplatesAdmin {
+		task ("Given a Page Template creator has a new Display Page Template") {
+			task ("Add a display page template for Web Content Article and Basic Web Content") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/PageTemplatesAdmin.testcase:55]: No need task block

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/PageTemplatesAdmin.testcase [hunk 21]
```diff
@@ -48,4 +48,147 @@ definition {
...
+
+				DisplayPageTemplatesAdmin.openDisplayPagesAdmin(siteURLKey = "test-site-name");
+
+				DisplayPageTemplatesAdmin.gotoDisplayPage(displayPageName = "Display Page Name");
+
+				PageEditor.clickPublish();
+			}
+		}
+
+		task ("When the Page Template creator open the ellipsis dropdown menu of display page template") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/PageTemplatesAdmin.testcase:70]: When the Page Template creator open the ellipsis menu of display page template

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagesadmin/PagesAdmin.testcase [hunk 22]
```diff
@@ -1396,6 +1396,83 @@ definition {
...
+				groupName = "Test Site Name",
+				layoutName = "Content Page Name",
+				type = "content");
+
+			JSONLayout.publishLayout(
+				groupName = "Test Site Name",
+				layoutName = "Content Page Name");
+		}
+
+		task ("When the Page designer open the ellipsis dropdown menu of content page") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagesadmin/PagesAdmin.testcase:1465]: When the Page designer open the dropdown menu of content page

**Review by Tim-Cao — COMMENTED** [review]
Hi @ozysouza, would you mind updating according to comments? Thanks

File: portal-web/test/functional/com/liferay/portalweb/macros/LexiconTable.macro [hunk 23]
```diff
@@ -1,5 +1,15 @@
 definition {
 
+	macro checkTableCheckbox {
+		Check(
+			key_tableEntry = "${tableEntry}",
+			locator1 = "LexiconTable#TABLE_ENTRY_CHECKBOX");
+
+		AssertChecked.assertCheckedNotVisible(
+			key_tableEntry = "${tableEntry}",
+			locator1 = "LexiconTable#TABLE_ENTRY_CHECKBOX");
```
[the remark this answers is not in the document]
Comment (ozysouza) [reply portal-web/test/functional/com/liferay/portalweb/macros/LexiconTable.macro:10]: I'll remove it, thanks for the explanation.

**Comment by ozysouza** [comment]
ci:test:sf

**Comment by ozysouza** [comment]
ci:forward

**Pull request closed without merging**